File: data/code_pairs/cpp/290.cpp
#include <iostream>
#include <vector>
#include <string>

class ByteType {
    char singleByte;
public:
    ByteType() : singleByte(0) {}
    ByteType(char c) : singleByte(c) {}
    char getByte() const { return singleByte; }
};

std::vector<ByteType> StringToBytes(const std::string &v) {
    std::vector<ByteType> bytes;
    for (char c : v) {
        bytes.push_back(ByteType(c));
    }
    return bytes;
}

std::string BytesToString(const std::vector<ByteType> &bytes) {
    std::string str;
    for (const ByteType &b : bytes) {
        str += b.getByte();
    }
    return str;
}

int main() {
    std::string str = "Hello, world!";
    auto bytes = StringToBytes(str);
    std::string convertedStr = BytesToString(bytes);

    if (convertedStr != str) {
        std::cerr << "Conversion error." << std::endl;
        return 1;
    }

    std::cout << "Conversion successful: " << convertedStr << std::endl;
    return 0;
}

File: data/code_pairs/fortran/290.f90
module TransferBug
    implicit none

    type :: ByteType
        character(len=1) :: singleByte
    end type ByteType

contains

    function StringToBytes(v) result(bytes)
        character(len=*), intent(in) :: v
        type(ByteType), allocatable :: bytes(:)
        integer :: i

        allocate(bytes(len(v)))
        do i = 1, len(v)
            bytes(i)%singleByte = v(i:i)
        end do
    end function StringToBytes

    subroutine BytesToString(bytes, string)
        type(ByteType), intent(in) :: bytes(:)
        character(len=*), intent(out) :: string
        integer :: numChars, i

        numChars = size(bytes)
        string = ''
        do i = 1, numChars
            string(i:i) = bytes(i)%singleByte
        end do
    end subroutine BytesToString

end module TransferBug

program main
    use TransferBug
    implicit none
    
    character(len=100) :: str = "Hello, world!"
    character(len=100) :: resultStr

    call BytesToString(StringToBytes(str), resultStr)

    if (trim(resultStr) /= trim(str)) then
        print *, "Conversion Failed"
    else
        print *, "Conversion Successful: ", trim(resultStr)
    end if
end program main